### PR TITLE
fix: On windows if a file is read only the cleanup reports errors.

### PR DESCRIPTION
**Pull request by grahamhar**
Adding the onerror to change the file permission then attempt the delete again fixes the issue reported in #37.

### CODE REVIEW

File: tftest.py [hunk 1]
```diff
@@ -298,6 +299,12 @@ def __init__(self, tfdir, basedir=None, binary='terraform', env=None):
   @classmethod
   def _cleanup(cls, tfdir, filenames, deep=True):
     """Remove linked files, .terraform and/or .terragrunt-cache folder at instance deletion."""
+
+    def remove_readonly(func, path, excinfo):
+      _LOGGER.warning(f'Issue deleting file {path}, caused by {excinfo}')
+      os.chmod(path, stat.S_IWRITE)
```
Comment (ludoo) [inline tftest.py:305]: would it make sense to catch errors inside this too?

**Comment by ludoo** [comment]
Thanks a lot for this! Can you sign the CLA, then it's good to go. Nitpick: does it make sense to exit cleanly from `remove_readonly` in case of errors there?

**Comment by ludoo** [comment]
Merging this, do you need a release on Pypi?

File: tftest.py (hunk 1 above)
Comment (grahamhar) [reply tftest.py:305]: I did think about this, but I think the purpose is only to try to fix a single issue and any other issues should be allowed to be handled upstream as they were before.

**Comment by grahamhar** [comment]
> Merging this, do you need a release on Pypi?

At your convenience that would be great.

**Comment by ludoo** [comment]
Here you go :) https://pypi.org/project/tftest/1.6.1/

**Pull request merged**